test/cloudtest/test_storaged.py

```diff
@@ -48,6 +48,49 @@ def test_storaged_creation(mz: MaterializeApplication) -> None:
         wait(condition="condition=Ready", resource=storaged)
 
 
+def test_storaged_resizing(mz: MaterializeApplication) -> None:
```

Comment:
@philip-stoev - Hi! Here's the `cloudtest` you offered to help me with last week. The test is currently failing; right now I think it's correctly showing some problem with the implementation, though I'm not sure.

Does this look reasonable? (And is there any special reason that changing a pod's resource allocation wouldn't cause it to be replaced in KinD vs. production? I notice that `.spec.containers.resources` is empty, though it seems like the code at least attempts to configure something there.)

Reply:
Give me a second to try it.

Reply:
I think the test is legit, however the storaged pod does not end up being restarted for some reason. `kubectl describe pod/storage-u1-0` reports no restarts and also it reports that the command-line arguments passed to `storaged` are still `--workers=1` and not the larger number desired. I see the following lines in the `environmentd` pod's log:

```
2022-08-15T17:07:34.825393Z  INFO mz_adapter::catalog: update source materialize.public.source1 (u1)
2022-08-15T17:07:34.876984Z  INFO mz_storage::controller::hosts: reprovisioning existing service storage-u1-0.storage-u1.default.svc.cluster.local:2100 for id u1
2022-08-15T17:07:40.216622Z  INFO mz_adapter::catalog: update source materialize.public.source1 (u1)
2022-08-15T17:07:40.260258Z  INFO mz_storage::controller::hosts: reprovisioning existing service storage-u1-0.storage-u1.default.svc.cluster.local:2100 for id u1
```

So for some reason, the "reprovisioning" line is printed twice and yet there are no other log lines that indicate that storaged has exited, has been restarted and has reconnected to the environmentd.

Reply:
I am not sure about the effect of modifying `resources`. In the KinD environment at least, the sizing is determined by the `StorageHostSizeMap` in `./src/adapter/src/catalog/config.rs`. There, the different sizes are differentiated only by their `--workers` command-line option.

Reply:
Thanks for taking a look!

> I am not sure about the effect of modifying resources. In the KinD environment at least, the sizing is determined by the StorageHostSizeMap in ./src/adapter/src/catalog/config.rs. There, the different sizes are differentiated only by their --workers command-line option.

Ah, you're right. I'm not sure why I thought there were different resource limits for each size, but the code is pretty clear here.

> kubectl describe pod/storage-u1-0 reports no restarts and also it reports that the command-line arguments passed to storaged are still `--workers=1` and not the larger number desired.

Great, that matches what I'm seeing locally as well.

I guess this behaviour is consistent with a) the client not actually making the request to kubernetes, and b) kubernetes not restarting a service which has only had its command line changed. And I suspect a) is more likely. Does that scan?

Reply:
I tried to dump the K8s resources for storaged to yaml , modified the --workers to 2 and then used `kubectl apply` to naively attempt to get it to pick the updated command line. I got the following error:

```
 spec: Forbidden: pod updates may not change fields other than `spec.containers[*].image`, `spec.initContainers[*].image`, `spec.activeDeadlineSeconds`, `spec.tolerations` (only additions to existing tolerations) or `spec.terminationGracePeriodSeconds` (allow it to be set to 1 if it was previously negative)
```

Which may indicate that one needs to destroy the existing service and pod and start them anew(?). But I am out of my element here, the #eng-cloud-observability channel may know best.

Reply:
If ALTER will be a DROP + CREATE of some sort, please note this ticket:
- #14188

Reply:
Yeah, this seems like KinD behaving as expected in any case... thanks!